packages/flutter_tools/lib/src/isolated/native_assets/ios/native_assets.dart

```diff
@@ -230,7 +230,7 @@ Future<void> _copyNativeAssetsIOS(
       }
       await lipoDylibs(dylibFile, sources);
       await setInstallNameDylib(dylibFile);
-      await createInfoPlist(targetUri.pathSegments.last, frameworkDir);
+      await createInfoPlist(targetUri.pathSegments.last, frameworkDir, minimumIOSVersion: '12.0');
```

Comment:
I believe this value should come from somewhere.

(Also this value should be passed into the `BuildConfig` so that the build hook can fail if it cannot produce a dylib with the right iOS version (https://github.com/dart-lang/native/issues/1133). This can be done in a separate PR though.)